### DISCUSSION TITLE
are these actually useable (SOFT CODING)

### DISCUSSION THREAD

**Discussion by RealDealJinx** [YoshiCrafter29/YoshiCrafterEngine#89] [Q&A]
i know this engine is new but is their a way to do either these 4

1: camera shaking mid song

2: modcharts (moving arrows up and down)

3: stage changing mid song

4: char trail + char floating

**Comment by RealDealJinx** [comment]
i forgot health drain

**Reply by RealReal334** [reply]
i think this engine reads .hx file and all of the above is possible

example character float


// This code will make your character levitate.

// Original character Y position
var ogY:Null<Int> = null;

// Angle of levitation (in radians)
var levitatingVariable = 0;
function update(elapsed:Float) {
    
    // Updates the variable by adding 180° per second.
    levitatingVariable += (elapsed * Math.PI);

    // If the original Y position is null, set it to the character's actual Y position.
    if (ogY == null) ogY = character.y;
    
    // Sets the character's Y position
    character.y = ogY - 200 + (Math.sin(levitatingVariable) * 100);
    
    // Set the character's angle.
    character.angle = Math.sin(levitatingVariable / 4) * 10;
}

**Reply by RealReal334** [reply]
https://www.youtube.com/watch?v=7BNe1LKgaJk
